### PR TITLE
Bug 2038419: Bump Grafana to 8.3.4 and handle datasource migration

### PR DESCRIPTION
**Pull request by arajkumar**
Newer version of grafana(e.g v8.3.3) seem to be supporting basic auth
only through `secureJsonData.basicAuthPassword`. This PR ensures the
creation of new datasources secret with `secureJsonData.basicAuthPassword` field and removes old secret during upgrade.

Signed-off-by: Arunprasad Rajkumar <arajkuma@redhat.com>

* [x] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Comment by arajkumar** [comment]
/retitle Bug 2038419: Bump Grafana to 8.3.4 and handle datasource migration

File: pkg/manifests/manifests.go [hunk 1]
```diff
@@ -276,6 +276,7 @@ var (
 	TrustedCABundleKey = "ca-bundle.crt"
 
 	AlertmanagerLegacyServiceMonitorName                = "alertmanager"
+	GrafanaLegacyDataSourceSecretName                   = "grafana-datasources"
```
Comment (simonpasquier) [inline pkg/manifests/manifests.go:279]: (nit) since it's only used in pkg/tasks/grafana.go, it could be moved there.

**Review by simonpasquier — COMMENTED** [review]
/lgtm
/hold

waiting for @prashbnair's review :)

**Review by PhilipGough — APPROVED** [review]
/lgtm

**Comment by arajkumar** [comment]
I've verified the upgrade path 4.9.x to 4.10.ci works with the fix.

File: pkg/tasks/grafana.go [hunk 2]
```diff
@@ -120,6 +122,11 @@ func (t *GrafanaTask) create(ctx context.Context) error {
 		return errors.Wrap(err, "reconciling Grafana Datasources Secret failed")
 	}
 
+	err = t.client.DeleteSecretByNamespaceAndName(ctx, sds.GetNamespace(), grafanaLegacyDataSourceSecretName)
```
Comment (prashbnair) [inline pkg/tasks/grafana.go:125]: Can we not reuse the DeleteSecret method by creating a Secret with name and namespace?

Comment (arajkumar) [reply pkg/tasks/grafana.go]: Fixed it.

/hold for testing!

**Pull request merged**